hatch/src/deps/mod.rs

```diff
@@ -68,23 +74,44 @@ pub fn clone_project_deps(path: &Path, user_defined_deps: &Vec<Dependency>) -> H
     Ok(())
 }
 
-fn clone_nested_project_deps(
-    registry: &Path,
-    path: &Path,
-    errored: &mut Vec<HatchError>,
-    visited: &mut HashSet<String>,
-) {
-    match task::read_project(path) {
-        Err(e) => {
-            errored.push(e);
-        }
-        Ok(current_project) => {
-            if !visited.contains(&current_project.name().to_owned()) {
-                current_project.deps().iter().for_each(|dep| {
-                    clone_dep(&dep.url(), &registry.join(dep.name()).as_path());
-                });
-                let _ = visited.insert(current_project.name().to_owned());
-            }
-        }
+fn clone_nested_project_deps(registry: &Path,
+                             path: &Path,
+                             errored: &mut Vec<HatchError>,
+                             visited: &mut HashSet<String>)
+{
+
+  // TODO: Do one of these:
+  // 1. Put this back in task::read_project
+  // 2. Make a constructor for Project that takes a &Path
+  let mut data = String::new();
+  let file = File::open(&path);
+  let _ = file.unwrap().read_to_string(&mut data);
+  match serde_yaml::from_str::<Project>(&data) {
```

Comment:
I've noticed this is the second time we have written out this whole process of opening the file and constructing a Project.  Should we encapsulate this somewhere?  *reads comment above it* we agree ;).  I would probably add a static constructor on Project named from_file(path: Path&) -> HatchResult<Project> since it can fail.